### PR TITLE
chore: Fix compiler warnings in consensus-service and consensus-service-impl

### LINKED ISSUE #14686: Fix compiler warnings in hedera-consensus-service

### PR DESCRIPTION
**Pull request by petreze**
**Description**:
Removed `tasks.withType<>...` line from `build.gradle.kts` files in `hedera-consensus-service` and `hedera-consensus-service-impl` which disables warnings for those packages.

- In service-impl `module-info.java` file java.compiler is changed from `static` to `transitive` as well as the export of 

**Related issue(s)**:

Fixes #14686, #14687 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicTest.java [hunk 1]
```diff
@@ -91,8 +91,8 @@ class ConsensusCreateTopicTest extends ConsensusTestBase {
     @Mock
     private EntityNumGenerator entityNumGenerator;
 
-    private WritableTopicStore topicStore;
-    private Configuration config;
+    private WritableTopicStore writableTopicStore;
+    private Configuration configuration;
```
Comment (mhess-swl) [inline hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicTest.java:95]: Are these necessary? They're fine, but...necessary?

File: config/checkstyle/checkstyle-suppressions.xml [hunk 2]
```diff
@@ -11,6 +11,9 @@
     <suppress files="[/\\]src[/\\].*[Tt]est[/\\](java|groovy)[/\\]" checks="VariableDeclarationUsageDistance" />
     <suppress files="[/\\]src[/\\].*[Tt]est[/\\](java|groovy)[/\\]" checks="VisibilityModifier" />
     <suppress files="[/\\]src[/\\].*[Tt]est[/\\](java|groovy)[/\\]" checks="AvoidStaticImport" />
+    <suppress files="[/\\]src[/\\].*[Tt]est[/\\](java|groovy)[/\\]" checks="AbbreviationAsWordInName" />
+    <suppress files="[/\\]src[/\\].*[Tt]est[/\\](java|groovy)[/\\]" checks="MemberName" />
+    <suppress files="[/\\]src[/\\].*[Tt]est[/\\](java|groovy)[/\\]" checks="MethodName" />
```
Comment (mhess-swl) [inline config/checkstyle/checkstyle-suppressions.xml:16]: Again, I'd prefer to isolate these to single spots if possible. See comment [here](https://github.com/hashgraph/hedera-services/pull/14685#discussion_r1709811671) for reference

Comment (petreze) [reply config/checkstyle/checkstyle-suppressions.xml]: Same as [here](https://github.com/hashgraph/hedera-services/pull/14685#discussion_r1711101452)

File: hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicTest.java (hunk 1 above)
Comment (petreze) [reply hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicTest.java]: https://github.com/hashgraph/hedera-services/pull/14685#discussion_r1711101452

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/ReadableTopicStoreImpl.java [hunk 3]
```diff
@@ -57,14 +57,14 @@ public ReadableTopicStoreImpl(@NonNull final ReadableStates states) {
      */
     @Override
     @Nullable
-    public Topic getTopic(@NonNull final TopicID id) {
+    public Topic getTopic(@NonNull TopicID id) {
```
[the remark this answers is not in the document]
Comment (petreze) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/ReadableTopicStoreImpl.java]: https://github.com/hashgraph/hedera-services/pull/14685#discussion_r1711101452

**Review by mhess-swl — APPROVED** [review]

**Review by derektriley — APPROVED** [review]
LGTM.

**Review by rbarkerSL — APPROVED** [review]

**Pull request merged**